presto-main/src/main/java/io/prestosql/operator/scalar/MathFunctions.java

```diff
@@ -607,11 +607,47 @@ public static long random(@SqlType(StandardTypes.BIGINT) long value)
         return ThreadLocalRandom.current().nextLong(value);
     }
 
+    @Description("a pseudo-random number between m and n (exclusive")
+    @ScalarFunction(value = "random", alias = "rand", deterministic = false)
+    @SqlType(StandardTypes.TINYINT)
+    public static long rangeRandomTinyInt(@SqlType(StandardTypes.TINYINT) long initialValue, @SqlType(StandardTypes.TINYINT) long finalValue) {
+
+        checkCondition(initialValue <= finalValue, INVALID_FUNCTION_ARGUMENT, "origin must be less than bound");
+        return ThreadLocalRandom.current().nextInt((int) initialValue, (int) finalValue);
```

Comment:
Is `nextInt` faster than `nextLong`?

If not, there is no need to cast to `int`.

Reply:
I've made some tests using `nanoTime` and I discover that `nextLong` is much more faster than `nextInt`, so I decided to change to `nextLong` and remove casts.